Feat: 장바구니 모달 컴포넌트 마크업/스타일링 구현

heydoopal opened the pull request:
## 🔧 반영 브랜치

feat/component-cart-modal-> develop

## 💬 Description
- 마크업/스타일링 구현 
  - stepper, button 컴포넌트 develop 브랜치에 merge 한 후 해당 컴포넌트에 삽입 필요해요!
- 장바구니 모달 open/close 로직 
- 상품 데이터 임포트 로직 구현
- 아직 컴포넌트 / 함수 관련 JS는 미완상태


Related to #5

heydoopal committed:
Feat: 장바구니 모달 컴포넌트 마크업/스타일링 구현

- 추후 stepper, button 로 교체

heydoopal committed:
Feat: comma 함수 구현

- 숫자 세 자리마다 콤마(',')를 생성해주는 함수 구현

heydoopal committed:
Feat: 상품 데이터 임포트 로직 구현

- 추후 포켓베이스에서 DB 삽입 후 작업 시 추출하여 사용할 경우 대비하여 테스트 겸 로직 작성해 보았습니당

heydoopal committed:
Refactor: 웹 접근성 개선 및 마크업 리팩토링

- HTML 문법검사 후 수정
- 웹 접근성 향상을 위한 속성 추가
- 클래스명

heydoopal committed:
Feat 장바구니 모달 open/close 로직 구현

- 근데 이걸 어떻게 컴포넌트화? 함수화? 시켜야할지 도저히 모르겠다
  - 다른데서 그냥 openModal(); 이렇게 쓰고싶은데!

heydoopal committed:
Design: 장바구니 모달 스타일링 수정

jaehwan-space reviewed (changes requested):
# 코드 리뷰 (PR Review)

1. 디자인 시안과 `font-weight` 일치 여부 확인

    - 디자인 시안과 일치하지 않는 font-weight를 확인해야 합니다. 

2. CSS 및 JS 파일 로드 최적화 및 모듈 활용

    - CSS 및 JS 파일을 로드할 때 모듈 시스템을 활용하여 필요한 부분만 로드하도록 최적화해야 합니다. 이는 성능을 향상시키고 코드 관리를 용이하게 합니다.
    ```javascript
    // 예시: 모듈을 이용한 최적화 / 필요한 모듈만 로드하여 사용
    import 'src/components/cartModal/cartModal.css';
    import 'src/components/cartModal/data.js';
    ```
3. cartModal.css에서 사용한 * 선택자에 대한 리스크

    - cartModal.css에서 사용한 `*` 선택자는 모든 요소에 스타일을 적용할 수 있어서 컴포넌트 조립 시 다른 요소들에도 영향을 미칠 수 있습니다. 이는 추후에 문제가 될 수 있으므로 가능한 한 구체적인 선택자를 사용하거나 CSS 모듈화를 고려해야 합니다.

4. 전체적인 폴더 경로 변경에 따른 절대경로 수정

    - 전체적인 폴더 경로 변경에 따라 절대경로를 수정해야 합니다. 프로젝트 구조를 확인하고 필요한 파일들의 경로를 정확히 지정해야 합니다.
---
위의 사항들을 검토하고 수정하면 장바구니 모달 컴포넌트의 디자인 일치성과 성능 최적화가 개선될 것입니다.

감사합니다.

heydoopal commented:
코드 리뷰 감사합니다 👼

혹시 피그마 디자인 시안과 꼭 100% 일치하게 작업해야 하나요 ? 
시안을 참고하여 다른 디자인으로도 꾸며도 된다고 알고있었어서요!
거의 대부분을 피그마 시안대로 작업하고 있으나, 시안과 실제 쇼핑몰에서 적용되어있는 부분이 상이한 부분이 있었고 
구현해봤을 때 조금 더 어울리는 폰트 두께나 요소 간 약간의 마진은 조정하고 있었거든요!

jaehwan-space commented:
시안과 100% 일치하게 작업하지 않아도 된다는 의견에 저도 동의합니다!
다만 아래의 사진과 같이 상품 제목 부분의 글자 두께가 달라지면 더 강조되는 효과도 있고 시각적으로 느낌도 달라져서 제안드립니다.
  <img src="https://github.com/FRONTENDSCHOOL10/ssagwa-kurly/assets/32737360/876dbf07-1bd1-4821-8816-7e0999e01db7" width="50%"> 
  <img src="https://github.com/FRONTENDSCHOOL10/ssagwa-kurly/assets/32737360/30acf9d3-dc42-4ea4-83a0-2bcb555ca2fc" width="50%">

heydoopal committed:
Merge branch 'develop' into feat/component-cart-modal

heydoopal commented:
앗! 저렇게 weight을 달리주니 제목이 더 강조가 되어서 좋은 것 같습니다! 👍  

그렇다면 제목에 font-weight을 더 주는게 좋다는 의견이신건가요!
리뷰에 시안과 다르게 작업해주셨다는 의견을 남겨주셔서요!
디자인 시안에는 모두 font-weight 600으로 되어있습니다!

heydoopal commented:
아아 제목에 강조를 주니 훨씬 더 좋은 것 같습니다! 👍

그럼 title에는 weight 값을 더 주는것이 좋다는 의견을 주신게 맞으실까요! 
figma시안에는 최종 금액 이외에 제목, 가격 등등 모두  600으로 되어있는데, (저는 모두 700으로 줬었네요!)
제목만 더 강조를 주는 걸로 수정하면 괜찮을까요 ㅎㅎ

jaehwan-space commented:
네네 그렇게 수정하면 좋을 것 같습니다! 
이외에도 유진님이 보시고 디자인적으로 더 괜찮을 것 같은 부분이 있다면 해당 부분도 수정 부탁드려요!

heydoopal committed:
Merge branch 'develop' into feat/component-cart-modal

heydoopal committed:
Modify: vite 설정 변경에 따른 경로 수정

heydoopal committed:
Modify: vite 설정 변경에 따른 경로 수정

heydoopal committed:
Modify: CSS, JS 파일 모듈 방식으로 임포트

heydoopal committed:
Design: 일부 스타일링 변경

- PR 리뷰에 따른 font-weight 설정 변경
- '*' 전체 선택자에서 body, button 태그 적용으로 변경
- 일부 margin 등 약간의 CSS 변경

heydoopal committed:
Merge branch 'develop' into feat/component-cart-modal

heydoopal committed:
Modify: 버튼 컴포넌트 적용

- 닫기 / 장바구니 담기 버튼에 버튼 컴포넌트 적용

heydoopal commented:
리뷰 주신내용 모두 수정 후 push하였습니다! :D

jaehwan-space reviewed (approved):
# 코드 리뷰 (PR Approve)
모든 수정사항 변경 확인했습니다.
